openpype/hosts/maya/plugins/create/create_multishot_layout.py

```diff
@@ -0,0 +1,206 @@
+    get_folder_by_name,
+    get_folder_by_path,
+    get_folders,
+)
+from maya import cmds  # noqa: F401
+
+from openpype import AYON_SERVER_ENABLED
+from openpype.client import get_assets
+from openpype.hosts.maya.api import plugin
+from openpype.lib import BoolDef, EnumDef, TextDef
+from openpype.pipeline import (
+    Creator,
+    get_current_asset_name,
+    get_current_project_name,
+)
+from openpype.pipeline.create import CreatorError
+
+
+class CreateMultishotLayout(plugin.MayaCreator):
+    """Create a multi-shot layout in the Maya scene.
+
+    This creator will create a Camera Sequencer in the Maya scene based on
+    the shots found under the specified folder. The shots will be added to
+    the sequencer in the order of their clipIn and clipOut values. For each
+    shot a Layout will be created.
+
+    """
+    identifier = "io.openpype.creators.maya.multishotlayout"
+    label = "Multi-shot Layout"
+    family = "layout"
+    icon = "project-diagram"
+
+    def get_pre_create_attr_defs(self):
+        # Present artist with a list of parents of the current context
+        # to choose from. This will be used to get the shots under the
+        # selected folder to create the Camera Sequencer.
+
+        """
+        Todo: get this needs to be switched to get_folder_by_path
+              once the fork to pure AYON is done.
+
+        Warning: this will not work for projects where the asset name
+                 is not unique across the project until the switch mentioned
+                 above is done.
+        """
+
+        current_folder = get_folder_by_name(
+            project_name=get_current_project_name(),
+            folder_name=get_current_asset_name(),
+        )
+
+        current_path_parts = current_folder["path"].split("/")
+
+        items_with_label = []
+        # populate the list with parents of the current folder
+        # this will create menu items like:
+        # [
+        #   {
+        #       "value": "",
+        #       "label": "project (shots directly under the project)"
+        #   }, {
+        #       "value": "shots/shot_01", "label": "shot_01 (current)"
+        #   }, {
+        #       "value": "shots", "label": "shots"
+        #   }
+        # ]
+
+        # go through the current folder path and add each part to the list,
+        # but mark the current folder.
+        for part_idx in range(len(current_path_parts)):
+            label = current_path_parts[part_idx]
+            if current_path_parts[part_idx] == current_folder["name"]:
+                label = f"{current_path_parts[part_idx]} (current)"
+            items_with_label.append(
+                dict(label=label,
+                     value="/".join(current_path_parts[:part_idx + 1]))
+            )
+        # add the project as the first item
+        items_with_label.insert(
+            0, dict(label=f"{self.project_name} "
+                          "(shots directly under the project)", value=""))
```

Comment:
If you want the project first - why don't append to put it in the list first?
Also use `enumerate` to get much more readable code for that for loop.
```suggestion
        # add the project as the first item
        items_with_label = [
            dict(label=f"{self.project_name} (shots directly under project)"
                 value="")
         ]

        # go through the current folder path and add each part to the list,
        # but mark the current folder.
        for part_idx, part in enumerate(current_path_parts):
            if part == current_folder["name"]:
                label = f"{part} (current)"
            else:
                label = part
            value = "/".join(current_path_parts[:part_idx + 1])
            items_with_label.append(
                dict(label=label,
                     value=value)
            )
```

Also, not sure if I'm a fan of the `dict(label=x, value=y)` over `{"label": x, "value": y}` but maybe it's just that I don't see that code style that much.

Reply:
seems like literal dictionary is much faster dict way, good to know.